logconsumer_test.go

```diff
@@ -0,0 +1,95 @@
+package testcontainers
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+type GetLogConsumer struct {
+	Msgs  []string
+	Await chan bool
+}
+
+func (g *GetLogConsumer) Accept(m string) {
+	g.Msgs = append(g.Msgs, m)
+	g.Await <- true
+}
+
+func Test_LogConsumerGetsCalled(t *testing.T) {
+	/*
+		send one request at a time to a server that will
+		print whatever was sent in the "echo" parameter, the log
+		consumer should get all of the messages and append them
+		to the Msgs slice
+	*/
+
+	ctx := context.Background()
+	req := ContainerRequest{
+		FromDockerfile: FromDockerfile{
+			Context:    "./testresources/",
+			Dockerfile: "echoserver.Dockerfile",
+		},
+		ExposedPorts: []string{"8080/tcp"},
+		WaitingFor:   wait.ForLog("ready"),
+	}
+
+	gReq := GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	c, err := GenericContainer(ctx, gReq)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ep, err := c.Endpoint(ctx, "http")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	g := GetLogConsumer{
```

Comment:
nit: rename struct to `TestLogConsumer`? It's used just here for the tests, right?
wdyt?

Reply:
You are correct, `TestLogConsumer` would be a better name. I will update, thanks!